project/src/pages/favorites/favorites.tsx

```diff
@@ -1,7 +1,45 @@
+import FavoritesItem from '../../components/favorites-item/favorites-item';
 import Footer from '../../components/footer/footer';
 import Header from '../../components/header/header';
+import {Offers, Offer} from '../../types/offers';
 
-function Favorites(): JSX.Element {
+type FavoritesProps = {
+  offers: Offers;
+}
+
+const isFavorites = (offer: Offer) => offer.isFavorite;
+
+const isOfferCityValue = (offer: Offer, city: string) => offer.city.name === city;
+
+const getOffersCityValue = (offers: Offers, city: string) => {
+  const groupOffersCity = new Array(0);
+  offers.forEach((offer) => {
+    if (isOfferCityValue(offer, city)) {
+      groupOffersCity.push(offer);
+    }
+  });
+  return groupOffersCity;
+};
+
+const getCityNames = (offers: Offers) => {
+  const arrayCityNames = new Array(0);
+  offers.forEach((offer: Offer) => arrayCityNames.push(offer.city.name));
+  return new Set(arrayCityNames);
+};
+
+const getGroupOffers = (offers: Offers) => {
+  const groupOffersCities = new Array(0);
+  const cityNames = getCityNames(offers);
+  cityNames.forEach((city) => groupOffersCities.push({
+    city: city,
+    offers: getOffersCityValue(offers, city),
+  }));
+  return groupOffersCities;
+};
+
+function Favorites(offers: FavoritesProps): JSX.Element {
+  const filterOffers = offers.offers.filter(isFavorites);
+  const groupOffers = getGroupOffers(filterOffers);
```

Comment:
Мне кажется больше подходит название 
```suggestion
  const groupOffers = mapOffersToCities(filterOffers);
```